cfme/tests/infrastructure/test_vm_reconfigure.py

```diff
@@ -329,10 +345,16 @@ def test_vm_reconfig_resize_disk_snapshot(disk_type, disk_mode, provider):
             1. VM selected
             2. Snapshot created
             3. Resize is not allowed when snapshots are attached
-    Bugzilla:
-        1631448
     """
-    pass
+    # Create snapshot
+    snapshot = new_snapshot(full_vm)
+    snapshot.create()
+
+    view = navigate_to(full_vm, 'Reconfigure')
+    row = [r for r in view.disks_table.rows()][0]
+    assert row.actions.widget.read() != "Resize"
+    with pytest.raises(NoSuchElementException):
+        row[9].widget.read()
```

Comment:
Q: what this mean? are we trying to read button?
I think when resize not allowed may be we can check like button is `disabled` or if button enabled but it will give flash message. I don't know exact behavior but I think this check is not sufficient.

Reply:
There is no way to check if button is disabled or enabled because the button is not visible at all, there is no other way to test the button. When the button is not present, `row.action.widget.read()` gives `Cancel` and `row[9].widget.read()` returns `NoSuchElementException` since there is no element present there. When the `Resize` button is visible, `row.action.widget.read()` gives `Resize` and `row[9].widget.read()` gives `Cancel`.

Reply:
`row[9].widget.is_displayed`?